src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_dedup_lazy_proxy.php

```diff
@@ -42,6 +42,13 @@ protected function createProxy($class, \Closure $factory)
         return $factory();
     }
 
+    protected function constructInstance($class, $proxy, ...$arguments)
+    {
+        get_parent_class($proxy) === $class ? method_exists($proxy, '__construct') && $proxy->__construct(...$arguments) : $proxy = new $class(...$arguments);
```

Comment:
should probably check `is_callable` too, to account for private constructors.

Reply:
Actually at this point having a private constructor would be a service definition error, no need to check for it on the proxy codepath while we don't for the code path that uses `new`. Better make them both fail in the usual loud way :)